realm/src/main/java/io/realm/internal/Table.java

```diff
@@ -491,71 +477,47 @@ public void addAt(long rowIndex, Object... values) {
             Object value = values[(int)columnIndex];
             switch (colTypes[(int)columnIndex]) {
             case BOOLEAN:
-                nativeInsertBoolean(nativePtr, columnIndex, rowIndex, (Boolean)value);
+                setBoolean(columnIndex, rowIndex, (Boolean)value);
```

Comment:
keep the call to the native methods here. Otherwise you call checkImmutable() multiple times.